tasks/monitoring.yml

```diff
@@ -0,0 +1,25 @@
+---
+- name: Fail if package state is not supported
+  fail:
+    msg: "The Cloud Ops Ansible role only supports the following package states: \"present\", and \"absent\"."
+  when: package_state not in ["present", "absent"]
+
+- block:
+  - name: Download script
```

Comment:
These tasks are not idempotent; they're going to show up as changed every time the role runs. That's considered bad practice (and will break check mode for this role). You need to refactor this to be idempotent (which probably means either modifying the shell script or replacing it with modules to directly modify the files).

Reply:
We initially decided to sacrifice idempotency for maintainability by delegating most of the logic out to the shell script. Mostly because we didn't want to have to manage custom installation logic across multiple config management tools in addition to the shell scripts.
I believe Ling and Igor are meeting later today to revisit that decision. If idemopotency is required I think using modules would be simpler than modifying the shell script.

Reply:
Modifying the shell script would probably be fairly straightforward; it would require adding a dry-run option of some kind that can be used in check mode, and either an exit status or a particular message that can be checked to indicate that no changes were made. Then you could reuse that across different config management tools.

I don't think the Ansible community would accept or use this role if it's not properly idempotent.

Reply:
IIUC the command module is skipped in check mode. If so, is the dry run option still necessary?

Reply:
Will address idempotency in a follow-up CL